### PR TITLE
*: refactors, docstrings, formatting, clippy fixes

### PR DESCRIPTION
**Pull request by steveej**
These changes shouldn't alter any functionality and are split off of #105 to keep that focused on the topic.

### CODE REVIEW

File: graph-builder/src/registry.rs [hunk 1]
```diff
@@ -280,13 +280,13 @@ pub fn fetch_releases(
 /// Update Images with release metadata should be immutable, but
 /// tags on registry can be mutated at any time. Thus, the cache
 /// is keyed on the hash of tag layers.
-fn cache_release(
+fn cache_release<S: ::std::hash::BuildHasher>(
```
Comment (lucab) [inline graph-builder/src/registry.rs:283]: This lint mostly makes sense when writing libraries with public API, where the consumer may need to pick the hasher. It isn't super-relevant in our case, I'd just allow it.

File: cincinnati/src/plugins/internal/edge_add_remove.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ impl PluginSettings for EdgeAddRemovePlugin {
     }
 }
 
-/// Adds and removes next and previous releases to/from quay labels
+/// Adds and removes next and previous releases specified by metadata
```
Comment (lucab) [inline cincinnati/src/plugins/internal/edge_add_remove.rs]: Missing final dot.

File: cincinnati/src/plugins/internal/edge_add_remove.rs [hunk 3]
```diff
@@ -50,7 +50,7 @@ impl EdgeAddRemovePlugin {
         Ok(Box::new(plugin))
     }
 
-    /// Remove next and previous releases from quay labels
+    /// Remove next and previous releases specified by metadata
```
Comment (lucab) [inline cincinnati/src/plugins/internal/edge_add_remove.rs]: Missing final dot.

File: cincinnati/src/plugins/internal/edge_add_remove.rs [hunk 4]
```diff
@@ -97,7 +97,7 @@ impl EdgeAddRemovePlugin {
         Ok(())
     }
 
-    /// Add next and previous releases from quay labels
+    /// Add next and previous releases specified by metadata
```
Comment (lucab) [inline cincinnati/src/plugins/internal/edge_add_remove.rs]: Missing final dot.

**Review by lucab — COMMENTED** [review]
Thanks a lot for the cleanup, looks very good to me. Just a bunch of very minor nits.

File: graph-builder/src/registry.rs (hunk 1 above)
Comment (steveej) [reply graph-builder/src/registry.rs:283]: I understand. However, [the caller of this function](https://github.com/openshift/cincinnati/pull/106/files#diff-991c40183d69ebfda48cb830e94be824R194), which is part of a public API, passes down a variable which has this signature.

Comment (lucab) [reply graph-builder/src/registry.rs:283]: I'm fine keeping the new form if you like it (it's good practice anyway).

Comment (steveej) [reply graph-builder/src/registry.rs:283]: My point was that we have to decide for the caller and the callee as they have to match. So if we want this form in the public API we have to use it internally too.

**Comment by steveej** [comment]
@lucab thanks for the review. I also noticed that I had an unintended left-over from the other PR in here which I removed now. Should be good for the final stamp now.

**Pull request merged**